internal/graph/cached_resolver_test.go

```diff
@@ -655,6 +655,32 @@ type document
 	require.Equal(t, uint32(1), res.GetResolutionMetadata().DatastoreQueryCount)
 }
 
+func TestCachedCheckResolver_ResolveCheck_After_Stop_DoesNotPanic(t *testing.T) {
+	cachedCheckResolver := NewCachedCheckResolver(WithExistingCache(nil)) // create cache inside
+
+	mockCtrl := gomock.NewController(t)
+	t.Cleanup(mockCtrl.Finish)
+
+	mockCheckResolver := NewMockCheckResolver(mockCtrl)
```

Comment:
Don't you also need to set the delegate of `mockCheckResolver`?

```suggestion
	mockCheckResolver := NewMockCheckResolver(mockCtrl)
	mockCheckResolver.SetDelegate(cachedCheckResolver)
```

Reply:
Hmmm no, it's irrelevant. I am mocking the scenario where the `localChecker` returns a terminal result that doesn't require further ecaluations.

Reply:
Then why instantiate the local checker at all?